### PR TITLE
#250: Fix initialization and comparison for non-HostSpace-accessible Kokkos containers

### LINKED ISSUE #250: Broken nvcc PR builds, possibly due to memory-space-sensitive Kokkos::View serialization
#202 may have broken PR testing for others, per @nlslatt

### PR DESCRIPTION
**Pull request by PhilMiller**
Fixes #250

### CODE REVIEW

**Comment by PhilMiller** [comment]
There's more to fix relating to `DynamicView`, but I want to get this in testing.

**Comment by PhilMiller** [comment]
The added support code in the test for `DynamicView` can go away with Kokkos 3.7, which has the corresponding `create_mirror_view_and_copy`, and may actually have that working.

File: tests/unit/test_kokkos_1d_commons.h [hunk 1]
```diff
@@ -62,36 +66,43 @@ static void compare1d(ViewT const& k1, ViewT const& k2) {
...
 }
 
+template <typename ViewT>
+static void compare1dDynamic(ViewT const& k1, ViewT const& k2) {
+  compareBasic(k1,k2);
+  std::cout << "compare1dDynamic: " << k1.extent(0) << "," << k2.extent(0) << "\n";
+  EXPECT_EQ(k1.extent(0), k2.extent(0));
+
+  for (typename ViewT::size_type i = 0; i < k1.extent(0); i++) {
+    EXPECT_EQ(k1.operator()(i), k2.operator()(i));
```
Comment (mzuzek) [inline tests/unit/test_kokkos_1d_commons.h:76]: Out of curiosity, why explicit `operator()` call instead of `EXPECT_EQ(k1(i), k2(i));` ?

**Review by mzuzek — APPROVED** [review]
Looks good!

File: tests/unit/test_kokkos_1d_commons.h (hunk 1 above)
Comment (PhilMiller) [reply tests/unit/test_kokkos_1d_commons.h:76]: Monkey see, monkey do. I was copying the existing code from `compareInner`

**Pull request merged**